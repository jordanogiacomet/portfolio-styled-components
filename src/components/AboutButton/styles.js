import styled, { css } from 'styled-components';

export const Container = styled.button`
  background: transparent;
  position: relative;
  border: none;
  color: #ADB7BE;
  margin-right: 12px;
  cursor: pointer;
  padding: 0.5rem 1rem;
  &.light{
    color: black;
    ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
  }
  &:hover {
    color: #1E90FF;
  }
  &::after {
    content: '';
    display: block;
    height: 2px;
    background: linear-gradient(to right, #1E90FF, #4169E1, #00008B);
    transition: width 0.3s ease-in-out;
    width: 0;
    position: absolute;
    left: 0;
    bottom: 0;
  }
  &:hover::after {
    width: 100%;
  }
  @media (min-width: 768px) {
    font-size: 2rem;
  }
  @media (max-width: 640px) {
    font-size: 1.6rem;
  }
  ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
`;
