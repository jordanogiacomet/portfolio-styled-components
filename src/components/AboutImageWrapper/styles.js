import styled from 'styled-components';

export const Container = styled.div`
  margin-top: 4px;
  order: 2;
  > img {
  display: block;
  max-width: 100%;
  height: auto;
  border-radius: 10px; /* Ajuste o raio da borda conforme necess√°rio */
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1), 0 6px 20px rgba(0, 0, 0, 0.1);
  }
  
  @media (min-width: 768px) {
  margin-top: 0;
  }
`;
