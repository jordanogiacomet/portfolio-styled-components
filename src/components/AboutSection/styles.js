import styled from 'styled-components';

export const Container = styled.section`
  color: white;
  padding: 50px 0px;
  display: flex;
  flex-direction: column;
  @media (min-width: 768px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 50px;
    align-items: center;
    margin-top: 80px;
  }
  @media (min-width: 1280px) {
    gap: 50px;

  };
`;
