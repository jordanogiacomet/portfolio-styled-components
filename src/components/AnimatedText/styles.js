import styled from 'styled-components';
import { TypeAnimation } from 'react-type-animation';

export const Container = styled(TypeAnimation)`
  display: inline-block;
  
  &.light{
    color: ${({ theme }) => theme.colors.primaryColor}
  }
  &.dark{
    ${({ theme }) => theme.colors.secondaryColor}
  }
`;