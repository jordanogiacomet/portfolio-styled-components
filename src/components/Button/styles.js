import styled, { css } from 'styled-components';

const getButtonStyles = ({ theme, type }) => {
  const buttonType = theme.button[type] || theme.button.default;
  return css`
    padding: ${buttonType.padding};
    border-radius: ${buttonType.borderRadius};
    color: ${buttonType.textColor};
    background: ${buttonType.background};
    font-weight: bold;
    cursor: pointer; 
    display: inline-flex;
    align-items: center;
    justify-content: center;
    min-width: 150px; /* Ajuste a largura mínima conforme necessário */
    transition: scale ease-in-out 0.2s;
    &:hover {
      transition: scale ease-in-out 0.2s;
      scale: calc(1.03);
    };

    @media (${theme.media.lteMedium}) {
      width: 100%; /* Largura 100% em telas menores */
    }
  `;
};

export const Container = styled.button`
  ${getButtonStyles};
`; 