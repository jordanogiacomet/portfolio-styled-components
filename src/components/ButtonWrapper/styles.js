import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  gap: 1rem;
  flex-wrap: wrap;
  margin-top: 2rem;

  @media (max-width: 768px) {
    flex-direction: column;
  }
`;