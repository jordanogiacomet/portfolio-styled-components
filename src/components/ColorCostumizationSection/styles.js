// src/components/ColorCustomizationSection/styles.js
import styled from 'styled-components';

export const Container = styled.section`
  display: flex;
  flex-direction: column;
  gap: 50px;
  margin-bottom: 50px;
`;
