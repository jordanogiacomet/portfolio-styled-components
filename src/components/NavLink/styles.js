import styled from 'styled-components';

export const Container = styled.a`
  display: block;
  padding-top: 0.5rem;  /* py-2 */
  padding-bottom: 0.5rem;  /* py-2 */
  padding-left: 0.75rem;  /* pl-3 */
  padding-right: 1rem;  /* pr-4 */
  color: #ADB7BE;  /* text-[#ADB7BE] */
  border-radius: 0.375rem;  /* rounded */
  text-decoration: none;
  transition: ease-in-out 0.3s;
  
  &.light{
    color: white;
    &:hover{
      color: #CCCCCC;
    }
  }
  
  &:hover{
    color: #4169E1;
  }
  @media (min-width: 768px) {
    margin-left: 15px;
  }

  @media (${({theme}) => theme.media.sm}) {  /* sm:text-xl */
    font-size: 1.8rem;
  }

  @media (${({theme}) => theme.media.md}) {  /* md:p-0 */
    padding: 0;
  }
`;
