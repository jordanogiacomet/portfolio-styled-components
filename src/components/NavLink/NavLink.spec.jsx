import { renderTheme } from '../../styles/render-theme';
import { NavLink } from '.';
import { screen } from '@testing-library/react';

describe('<NavLink />', () => {
  it('should render with children', () => {
    renderTheme(<NavLink href="http://localhost">Link</NavLink>);
    const link = screen.getByRole('link', { name: 'Link' });

    expect(link).toHaveAttribute('target', '_self');
  });
  it('should match snapshot', () => {
    const { container } = renderTheme(<NavLink href="http://localhost">Link</NavLink>);

    expect(container).toMatchSnapshot();
  });
});


