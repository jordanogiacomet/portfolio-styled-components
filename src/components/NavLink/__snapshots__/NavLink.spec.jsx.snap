// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLink /> should match snapshot 1`] = `
.c0 {
  display: block;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 0.75rem;
  padding-right: 1rem;
  color: #ADB7BE;
  border-radius: 0.375rem;
  text-decoration: none;
  transition: ease-in-out 0.3s;
}

.c0:hover {
  color: #4169E1;
}

@media (min-width: 768px) {
  .c0 {
    margin-left: 15px;
  }
}

@media ((min-width: 640px)) {
  .c0 {
    font-size: 1.8rem;
  }
}

@media ((min-width: 768px)) {
  .c0 {
    padding: 0;
  }
}

<div>
  <a
    class="c0"
    href="http://localhost"
    target="_self"
  >
    Link
  </a>
</div>
`;
